### PR TITLE
updated commands and permissions to match current klasa version

### PR DESCRIPTION
**Pull request by PyroTechniac**

### CODE REVIEW

**Review by kyranet — APPROVED** [review]
Nice, thanks you! Does Posrednik work in latest Klasa with this patch?

**Comment by PyroTechniac** [comment]
I will test it tonight and make and required changes!

**Comment by PyroTechniac** [comment]
That should be everything!

File: commands/Announcement/announcement.js [hunk 1]
```diff
@@ -6,20 +6,18 @@ module.exports = class extends Command {
...
             usage: '<string:string>',
-            cooldown: 60
+            cooldown: 60,
+            requiredSettings: ['channels.announcementChannel', 'roles.announcementRole']
         });
     }
 
     async run(msg, [message]) {
         const announcementID = msg.guild.settings.channels.announcementChannel;
-        if (!announcementID) throw msg.language.get('COMMAND_SUBSCRIBE_NO_CHANNEL');
```
Comment (kyranet) [inline commands/Announcement/announcement.js:19]: Why did you remove the checks?

File: monitors/antiinvite.js [hunk 2]
```diff
@@ -6,7 +6,9 @@ module.exports = class extends Monitor {
         super(...args, {
             name: 'antiinvite',
             enabled: true,
-            ignoreSelf: true
+            ignoreSelf: true,
```
Comment (kyranet) [inline monitors/antiinvite.js]: Remove enabled, ignoreSelf and ignoreEdits, they all default to true.

File: commands/Misc/echo.js [hunk 3]
```diff
@@ -14,8 +14,11 @@ module.exports = class extends Command {
     }
 
     async run(msg, [channel = msg.channel, ...message]) {
-        if (channel.postable === false) throw msg.language('SYSTEM_CHANNEL_NOT_POSTABLE');
```
Comment (kyranet) [inline commands/Misc/echo.js:17]: This does exist.

https://github.com/dirigeants/klasa/blob/628fa2b5d75e677510adb9f83b7ecfefd6677192/src/extendables/MessageExtendables.js#L18-L20

Comment (kyranet) [reply commands/Misc/echo.js:17]:
```javascript
if (!channel.postable) throw msg.language('SYSTEM_CHANNEL_NOT_POSTABLE');
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: commands/Announcement/announcement.js (hunk 1 above)
Comment (PyroTechniac) [reply commands/Announcement/announcement.js:19]: because required permissions will throw if they're not set, I can put them back if you would prefer however

Comment (kyranet) [reply commands/Announcement/announcement.js:19]: Yes, please.

Comment (kyranet) [reply commands/Announcement/announcement.js:19]: Actually, only this check should go. The other two are fine.

Comment (PyroTechniac) [reply commands/Announcement/announcement.js:19]: So for clarification, just re-add the check for announcement channel?

Comment (kyranet) [reply commands/Announcement/announcement.js:19]: And if it's postable. Because that check (requiredPermissions) checks for the value, but not if the resolved value exists.

File: commands/Misc/echo.js (hunk 3 above)
Comment (kyranet) [reply commands/Misc/echo.js:17]: Please change the `=== false` to `!`

**Review by kyranet — APPROVED** [review]
Thanks you!

**Comment by PyroTechniac** [comment]
No problem!
